File: 724-find-pivot-index/find-pivot-index.py
class Solution(object):
    def pivotIndex(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        global_sum = sum(nums)
        left_sum = 0

        for i, n in enumerate(nums):
            if global_sum - (2 * left_sum) == n:
                return i

            left_sum += n
        
        return -1

File: 724-find-pivot-index/find-pivot-index.ts
function pivotIndex(nums: number[]): number {
    const globalSum = nums.reduce((a,b) => a + b);
    let leftSum = 0; 

    for (let i = 0; i < nums.length; i++) {
        if (globalSum - (leftSum * 2) === nums[i]) {
            return i;
        }
        leftSum += nums[i];
    }
    return -1;
};